### PR TITLE
test(bats): added bats example for deleting an image

### LINKED ISSUE #1706: [Test] Add blackbox tests/examples for deleting images
There seems to be confusion for users trying to delete images on how they can use the dist-spec API to do so.
Let's add some examples / bats tests for this.

### PR DESCRIPTION
**Pull request by laurentiuNiculae**
**What type of PR is this?**
Closes #1706

**Which issue does this PR fix**:

**What does this PR do / Why do we need it**:

**If an issue # is not available please add repro steps and logs showing the issue**:

**Testing done on this change**:

**Automation added to e2e**:

**Will this break upgrades or downgrades?**

**Does this PR introduce any user-facing change?**:

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by andaaron — APPROVED** [review]
lgtm

File: test/blackbox/delete_images.bats [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    [ "$status" -eq 0 ]
+    run curl http://127.0.0.1:8080/v2/_catalog
+    [ "$status" -eq 0 ]
+    [ $(echo "${lines[-1]}" | jq '.repositories[]') = '"alpine"' ]
+
+    MANIFEST_DIGEST=$(skopeo inspect --tls-verify=false docker://localhost:8080/alpine:3.17.3 | jq -r '.Digest')
+    echo ${MANIFEST_DIGEST}
+}
+
+function teardown() {
```
Comment (peusebiu) [inline test/blackbox/delete_images.bats:56]: teardown_file() for consistency

File: test/blackbox/helpers_delete_images.bash [hunk 2]
```diff
@@ -0,0 +1,84 @@
+ROOT_DIR=$(git rev-parse --show-toplevel)
```
Comment (peusebiu) [inline test/blackbox/helpers_delete_images.bash:1]: can't we use an existing helper script?

Comment (laurentiuNiculae) [reply test/blackbox/helpers_delete_images.bash:1]: Yes, I think we can

Comment (laurentiuNiculae) [reply test/blackbox/helpers_delete_images.bash:1]: I tried to make it similar to the other tests

Comment (laurentiuNiculae) [reply test/blackbox/helpers_delete_images.bash:1]: I tried to use other helper scrips and none work for this bats tests, i'm not sure why

Comment (peusebiu) [reply test/blackbox/helpers_delete_images.bash:1]: ok

**Review by andaaron — APPROVED** [review]

**Comment by andaaron** [comment]
Waiting on @adodon2go to refactor the bats tests.

**Review by andaaron — DISMISSED** [review]
LGTM

File: test/blackbox/helpers_zot.bash [hunk 3]
```diff
@@ -25,6 +25,8 @@ function zot_stop_all() {
 function wait_zot_reachable() {
     local zot_port=${1}
     local zot_url=http://127.0.0.1:${zot_port}/v2/_catalog
+    echo "wait_zot_reachable INSIDE"
```
Comment (adodon2go) [inline test/blackbox/helpers_zot.bash:28]: Remove this lines

File: test/blackbox/delete_images.bats [hunk 4]
```diff
@@ -0,0 +1,127 @@
+load helpers_zot
```
Comment (adodon2go) [inline test/blackbox/delete_images.bats:1]: Add Makefile target + ci check

Comment (andaaron) [reply test/blackbox/delete_images.bats:1]: Will consolidate in https://github.com/project-zot/zot/issues/1743

File: Makefile [hunk 5]
```diff
@@ -406,6 +406,10 @@ test-bats-sync: check-linux binary binary-minimal bench check-skopeo $(BATS) $(N
 	$(BATS) --trace --print-output-on-failure test/blackbox/sync_docker.bats
 	$(BATS) --trace --print-output-on-failure test/blackbox/sync_replica_cluster.bats
 
+.PHONY: test-bats-delete-image
+test-bats-delete-image: check-linux binary binary-minimal bench check-skopeo $(BATS)
```
Comment (adodon2go) [inline Makefile:410]: Why do we need bench as prereq?
Also remove it from test-bats-sync

Comment (andaaron) [reply Makefile:410]: I will refactor this.

**Review by andaaron — APPROVED** [review]

**Pull request merged**